api-get-object.go

```diff
@@ -627,18 +627,21 @@ func (c Client) getObject(ctx context.Context, bucketName, objectName string, op
 	if err != nil {
 		return nil, ObjectInfo{}, nil, err
 	}
+
+	var header http.Header
 	if resp != nil {
 		if resp.StatusCode != http.StatusOK && resp.StatusCode != http.StatusPartialContent {
 			return nil, ObjectInfo{}, nil, httpRespToErrorResponse(resp, bucketName, objectName)
 		}
+		header = resp.Header
 	}
 
-	objectStat, err := ToObjectInfo(bucketName, objectName, resp.Header)
+	objectStat, err := ToObjectInfo(bucketName, objectName, header)
 	if err != nil {
 		closeResponse(resp)
-		return nil, objectStat, resp.Header, nil
+		return nil, objectStat, header, nil
```

Comment:
I found that there is situation when resp == nil,  calling `resp.Header` raise panic

Reply:
Wait it is guaranteed that resp is non nil if err is nil, I think there is some other bug internally which needs to be fixed.

This is introduced by this change @fortyanov

Reply:
Got it, revert tests back.

Reply:
Got it, revert tests back.